src/SprykerEco/Zed/AmazonQuicksight/Business/Deleter/QuicksightUserDeleter.php

```diff
@@ -0,0 +1,231 @@
+            return $this->executeDeleteQuicksightUsersByUserCollectionResponseTransaction(
+                $userCollectionResponseTransfer,
+                $filteredUserTransfers,
+            );
+        });
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\QuicksightUserCollectionResponseTransfer
+     */
+    public function deleteRegisteredQuicksightUsersNotMatchedWithExistingUsers(): QuicksightUserCollectionResponseTransfer
+    {
+        $quicksightUserCollectionResponseTransfer = new QuicksightUserCollectionResponseTransfer();
+
+        $quicksightListUsersResponseTransfer = $this->amazonQuicksightApiClient->listUsers();
+        if ($quicksightListUsersResponseTransfer->getErrors()->count() !== 0) {
+            return $quicksightUserCollectionResponseTransfer->setErrors(
+                $quicksightListUsersResponseTransfer->getErrors(),
+            );
+        }
+
+        $quicksightUserTransfers = $quicksightListUsersResponseTransfer->getQuicksightUsers();
+        if ($quicksightUserTransfers->count() === 0) {
+            return $quicksightUserCollectionResponseTransfer;
+        }
+
+        $notMatchedQuicksightUserTransfers = $this->quicksightUserMatcher->getQuicksightUsersNotMatchedWithExistingUsers(
+            $quicksightUserTransfers,
+        );
+
+        foreach ($notMatchedQuicksightUserTransfers as $quicksightUserTransfer) {
+            $quicksightUserCollectionResponseTransfer = $this->deleteQuicksightUser(
+                $quicksightUserTransfer,
+                $quicksightUserCollectionResponseTransfer,
+            );
+        }
+
+        return $quicksightUserCollectionResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UserCollectionResponseTransfer $userCollectionResponseTransfer
+     * @param array<int|string, \Generated\Shared\Transfer\UserTransfer> $userTransfers
+     *
+     * @return \Generated\Shared\Transfer\UserCollectionResponseTransfer
+     */
+    protected function executeDeleteQuicksightUsersByUserCollectionResponseTransaction(
+        UserCollectionResponseTransfer $userCollectionResponseTransfer,
+        array $userTransfers
+    ): UserCollectionResponseTransfer {
+        $userIdsToDelete = [];
+        foreach ($userTransfers as $entityIdentifier => $userTransfer) {
+            $quicksightDeleteUserResponseTransfer = $this->amazonQuicksightApiClient->deleteUserByUsername($userTransfer);
+            if ($quicksightDeleteUserResponseTransfer->getErrors()->count() !== 0) {
+                $userCollectionResponseTransfer = $this->addErrorsToUserCollectionResponse(
+                    $userCollectionResponseTransfer,
+                    $quicksightDeleteUserResponseTransfer->getErrors(),
+                    (string)$entityIdentifier,
+                );
+
+                $this->messengerFacade->addInfoMessage(
+                    (new MessageTransfer())->setValue(static::ERROR_MESSAGE_QUICKSIGHT_USER_DELETION_FAILED),
+                );
```

Comment:
not sure this is a part of delete transaction, maybe it is better to move to the parent method?

Reply:
I'd leave it here because it would be hard to distinguish errors that were added to `QuicksightUserCollectionResponseTransfer` in previous plugins and the ones we added here. 
So to not overcomplicate things, I'd leave this code here.